pkg/controllers/networkinfo/networkinfo_utils.go

```diff
@@ -51,6 +51,12 @@ func deleteSuccess(r *NetworkInfoReconciler, _ *context.Context, o *v1alpha1.Net
 }
 
 func setNetworkInfoVPCStatus(ctx *context.Context, networkInfo *v1alpha1.NetworkInfo, client client.Client, createdVPC *v1alpha1.VPCState) {
+	// if createdVPC is empty, remove the VPC from networkInfo
+	if reflect.DeepEqual(*createdVPC, v1alpha1.VPCState{}) {
```

Comment:
this is a specific scenario for update fail to remove all state, if so, we could give a nil value directly instead of comparing 2 empty obj.

Reply:
ok